src/discordtipbot/middleware/errorcatcher.cr

```diff
@@ -0,0 +1,8 @@
+class ErrorCatcher
+  def call(payload, context)
+    yield
+  rescue ex
+    context[Discord::Client].create_message(payload.channel_id, "There was an unexpected error. This has been reported and should be resolved soon") if payload.is_a?(Discord::Gateway::MessageUpdatePayload)
```

Comment:
.. `if payload.is_a?(Discord::Gateway::MessageUpdatePayload)` ? :confused:

Reply:
I want to call ErrorCatcher on all event handlers. Not only on_message, which is the only event it makes sense to reply to.